network/test/middleware_test.go

```diff
@@ -26,11 +28,31 @@ import (
 
 const testChannel = "test-channel"
 
+type TagsObserver struct {
```

Comment:
```suggestion
type tagsObserver struct {
```
Better not to export test data types.